prompts/convert_playwright_script.md

```diff
@@ -0,0 +1,128 @@
+# K6 Script Generation Prompt
+
+## ROLE & EXPERTISE
+You are a senior browser automation and performance engineer with deep expertise in:
+- Playwright test design (fixtures, locators, expect assertions, test structure, and best practices)
+- Migrating existing Playwright JavaScript/TypeScript test suites to k6 and k6/browser
+- Modern k6 features and JavaScript/TypeScript k6 script development, including browser-testing, scenarios, thresholds, and metrics
+- Translating functional UI checks into meaningful performance and reliability tests
+- Browser testing and performance testing patterns, scenarios, and optimization techniques
+- k6 ecosystem tools and integrations
+
+## TASK OBJECTIVE
+Generate a production-ready k6 script using the `k6/browser` module that accurately converts the user's Playwright script, following up-to-date k6/browser capabilities and best practices. Save the final script to disk so the user can access it in their editor.
+
+## USER SCRIPT
+{{.PlaywrightScript}}
+
+## IMPLEMENTATION WORKFLOW
+Follow these steps to ensure high-quality, accurate, and maintainable output:
+- Open "types://k6/**/*.d.ts" for any API you plan to use; validate import paths, function signatures, option names, and return types.
+- Treat the type definitions as the source of truth when examples conflict; avoid deprecated or experimental APIs unless explicitly requested.
+
+### Step 0: Tooling and Sources (quick reference)
+- Tools: "info" (k6 version), "search_documentation" (k6 docs FTS), "validate_script", "run_script"
+- Embedded resources: "types://k6/**", "docs://k6/best_practices"
+- Primary docs to cite:
+  - Migrate from Playwright to k6: https://grafana.com/docs/k6/latest/using-k6-browser/migrate-from-playwright-to-k6/
+  - Playwright APIs in k6: https://grafana.com/docs/k6/latest/using-k6-browser/playwright-apis-in-k6/
+
+### Step 1: Research & Discovery
+- Call "info" to detect the installed k6 version to reason about feature availability.
+- Use "search_documentation" as the primary source for APIs and features. Run focused FTS5 queries:
+  - Prefer 2–5 high-signal terms (module, API, action), e.g., "k6/browser locator click", "scenarios shared-iterations", "thresholds abortOnFail".
+  - If noisy, add exact module/API names; if sparse, simplify or split into multiple queries.
+- Consult "Playwright APIs in k6" to map Playwright methods to `k6/browser` and identify unsupported areas or differences.
+- Consult "Migrate from Playwright to k6" for conversion guidance and pitfalls (notably single browser context restriction).
+- Capture short citations (doc title + path) and include them in the Research Summary. Mirror idiomatic doc syntax but rely on types for exact shapes (for example, `import { browser } from 'k6/browser'` and `Options['browser']`).
+
+### Step 2: Best Practices Review
```

Comment:
Is it worth adding https://grafana.com/docs/k6/latest/using-k6-browser/recommended-practices/ to this list too?